### PR TITLE
feat: Add rule_type parameter to merge_request_approval_rule

### PR DESCRIPTION
**Pull request by petrmvala**
This is working in the API, and is necessary in order to enable
Eligible Approvers.

Closes: #904

### PR Checklist Acknowledgement

- [x] I acknowledge that all of the following items are true, where applicable:
  - Resource attributes match 1:1 the names and structure of the API resource in [the GitLab API documentation](https://docs.gitlab.com/ee/api/).
  - [Examples](https://github.com/gitlabhq/terraform-provider-gitlab/tree/main/examples) are updated with:
    - A \*.tf file for the resource/s with at least one usage example
    - A \*.sh file for the resource/s with an import example (if applicable)
  - New resources have at minimum a basic test with three steps:
    - Create the resource
    - Update the attributes
    - Import the resource
  - No new `//lintignore` comments were copied from existing code. (Linter rules are meant to be enforced on new code.)

### CODE REVIEW

File: internal/provider/resource_gitlab_project_approval_rule.go [hunk 1]
```diff
@@ -43,6 +49,13 @@ var _ = registerResource("gitlab_project_approval_rule", func() *schema.Resource
 				Type:        schema.TypeInt,
 				Required:    true,
 			},
+			"rule_type": {
+				Description:      fmt.Sprintf("String, defaults to 'regular'. The type of rule. `any_approver` is a pre-configured default rule with `approvals_required` at `0`. Valid values are %s.", renderValueListForDocs(validRuleTypeValues)),
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          "regular",
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project_approval_rule.go:56]: NIT:

```suggestion
```

I'd leave it up to GitLab to decide on the `rule_type` ... thus, we don't send it, if it wasn't set :)

**Review by timofurrer — CHANGES_REQUESTED** [review]
One minor thing, otherwise LGTM 🎉

File: internal/provider/resource_gitlab_project_approval_rule.go (hunk 1 above)
Comment (petrmvala) [reply internal/provider/resource_gitlab_project_approval_rule.go:56]: Ah, right. makes sense :)

**Review by timofurrer — DISMISSED** [review]
LGTM 🎉

File: internal/provider/resource_gitlab_project_approval_rule.go [hunk 2]
```diff
@@ -43,6 +49,12 @@ var _ = registerResource("gitlab_project_approval_rule", func() *schema.Resource
 				Type:        schema.TypeInt,
 				Required:    true,
 			},
+			"rule_type": {
+				Description:      fmt.Sprintf("String, defaults to 'regular'. The type of rule. `any_approver` is a pre-configured default rule with `approvals_required` at `0`. Valid values are %s.", renderValueListForDocs(validRuleTypeValues)),
+				Type:             schema.TypeString,
+				Optional:         true,
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project_approval_rule.go:55]:
```suggestion
				Optional:         true,
				Computed:      true,
```

**Review by timofurrer — APPROVED** [review]

**Comment by timofurrer** [comment]
@petrmvala I think we have a problem ...

This change doesn't handle the update at all. I've just double checked go-gitlab and the upstream API and neither support the *rule type* (see https://docs.gitlab.com/ee/api/merge_request_approvals.html#update-project-level-rule)

I think for now we have to mark it with [`ForceNew`](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#forcenew). That being said a test changing the rule type would be great.

**Review by timofurrer — CHANGES_REQUESTED** [review]
@petrmvala I think we have a problem ...

This change doesn't handle the update at all. I've just double checked go-gitlab and the upstream API and neither support the *rule type* (see https://docs.gitlab.com/ee/api/merge_request_approvals.html#update-project-level-rule)

I think for now we have to mark it with [`ForceNew`](https://www.terraform.io/plugin/sdkv2/schemas/schema-behaviors#forcenew). That being said a test changing the rule type would be great.

File: internal/provider/resource_gitlab_project_approval_rule_test.go [hunk 3]
```diff
@@ -79,15 +79,83 @@ func TestAccGitLabProjectApprovalRule_basic(t *testing.T) {
...
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectApprovalRuleExists("gitlab_project_approval_rule.bar", &projectApprovalRule),
+					testAccCheckGitlabProjectApprovalRuleAttributes_AnyApprover(&projectApprovalRule, &testAccGitlabProjectApprovalRuleExpectedAttributes_AnyApprover{
+						Name:              "bar",
+						ApprovalsRequired: 2,
+						RuleType:          "any_approver",
+					}),
+				),
+			},
+			// Re-create rule
```
Comment (timofurrer) [inline internal/provider/resource_gitlab_project_approval_rule_test.go:122]:
```suggestion
			// Update rule
```

**Review by timofurrer — APPROVED** [review]

**Pull request merged**